### PR TITLE
Add docs and message for DDP static graph

### LINKED ISSUE #12296: Document DistributedDataParallel Static Graph support
## 📚 Documentation

PyTorch 1.11 makes static graph a stable feature for DDP. https://pytorch.org/blog/pytorch-1.11-released/#distributed-training

There are no code changes needed to support this from the Lightning framework POV: all kwargs to the DDPStrategy are forwarded to the DistributedDataParallel constructor here: https://github.com/PyTorchLightning/pytorch-lightning/blob/3b4061f39a8af47a77e9dfa58256e818784b21ca/pytorch_lightning/strategies/ddp.py#L159

Users can enable this with the following
```py
trainer = Trainer(strategy=DDPStrategy(static_graph=True))
```

This should be added to the DDP optimization sections in the docs here: https://pytorch-lightning.readthedocs.io/en/latest/advanced/advanced_gpu.html#ddp-optimizations
______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @rohitgr7 @justusschock @kaushikb11 @awaelchli @akihironitta

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes #12296

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @rohitgr7 @justusschock @kaushikb11 @awaelchli @akihironitta

### CODE REVIEW

File: docs/source/advanced/model_parallel.rst [hunk 1]
```diff
@@ -737,9 +737,6 @@ DDP Communication Hooks
 
 DDP Communication hooks is an interface to control how gradients are communicated across workers, overriding the standard allreduce in DistributedDataParallel. This allows you to enable performance improving communication hooks when using multiple nodes.
 
-.. note::
-    DDP communication hooks needs pytorch version at least 1.8.0
```
Comment (carmocca) [inline docs/source/advanced/model_parallel.rst:741]: 1.8 is the min version now so we don't need this note

**Review by rohitgr7 — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
